In [55]:
import pandas as pd
import numpy as np

In [56]:
# YEAR 2000
df = pd.read_csv('Data/final_tmdb_data_2000.csv.gz')
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1235 entries, 0 to 1234
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1235 non-null   object 
 1   adult                  1234 non-null   float64
 2   backdrop_path          679 non-null    object 
 3   belongs_to_collection  113 non-null    object 
 4   budget                 1234 non-null   float64
 5   genres                 1234 non-null   object 
 6   homepage               64 non-null     object 
 7   id                     1234 non-null   float64
 8   original_language      1234 non-null   object 
 9   original_title         1234 non-null   object 
 10  overview               1212 non-null   object 
 11  popularity             1234 non-null   float64
 12  poster_path            1112 non-null   object 
 13  production_companies   1234 non-null   object 
 14  production_countries   1234 non-null   object 
 15  rele

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026  0.000  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092  0.000                               NaN                   NaN   
3  tt0116391  0.000                               NaN                   NaN   
4  tt0118694  0.000  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

        budget                                             genres homepage  \
0          NaN                                                NaN      NaN   
1 10000000.000  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2        0.000           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3        0.000  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4   150000.000  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

          id original_language   original_title  ...      revenue  runtime  \
0        NaN               NaN              NaN  ...          NaN      NaN   
1  62127.000                en  The Fantasticks  ...        0.000   86.000   
2 110977.000                en    For the Cause  ...        0.000  100.000   
3 442869.000                hi             Gang  ...        0.000  152.000   
4    843.000                cn             花樣年華  ... 12854953.000   99.000   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1  0.000         5.500     22.000           NaN  
2  0.000         5.100      8.000           NaN  
3  0.000         4.000      1.000           NaN  
4  0.000         8.115   2187.000            PG  

[5 rows x 26 columns]

In [57]:
# YEAR 2001
df2 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz')
df2.info()
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1328 entries, 0 to 1327
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1328 non-null   object 
 1   adult                  1327 non-null   float64
 2   backdrop_path          711 non-null    object 
 3   belongs_to_collection  93 non-null     object 
 4   budget                 1327 non-null   float64
 5   genres                 1327 non-null   object 
 6   homepage               109 non-null    object 
 7   id                     1327 non-null   float64
 8   original_language      1327 non-null   object 
 9   original_title         1327 non-null   object 
 10  overview               1298 non-null   object 
 11  popularity             1327 non-null   float64
 12  poster_path            1192 non-null   object 
 13  production_companies   1327 non-null   object 
 14  production_countries   1327 non-null   object 
 15  rele

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423  0.000  /hfeiSfWYujh6MKhtGTXyK3DD4nN.jpg                   NaN   
2  tt0114447  0.000                               NaN                   NaN   
3  tt0118589  0.000  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   
4  tt0118652  0.000  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

        budget                                             genres homepage  \
0          NaN                                                NaN      NaN   
1 48000000.000  [{'id': 10749, 'name': 'Romance'}, {'id': 14, ...      NaN   
2        0.000  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3 22000000.000  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   
4  1000000.000  [{'id': 27, 'name': 'Horror'}, {'id': 9648, 'n...      NaN   

          id original_language         original_title  ...      revenue  \
0        NaN               NaN                    NaN  ...          NaN   
1  11232.000                en         Kate & Leopold  ... 76019048.000   
2 151007.000                en       The Silent Force  ...        0.000   
3  10696.000                en                Glitter  ...  5271666.000   
4  17140.000                en  The Attic Expeditions  ...        0.000   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1  118.000  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2   90.000  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  104.000  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4  100.000  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century, they'd be p...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1  0.000         6.325   1185.000         PG-13  
2  0.000         5.000      3.000           NaN  
3  0.000         4.536    124.000         PG-13  
4  0.000         5.100     29.000             R  

[5 rows x 26 columns]

In [58]:
# Concatenate two dataframes 
df3 = pd.concat([df, df2], axis = 0)
df3.info()
df3.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2563 entries, 0 to 1327
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2563 non-null   object 
 1   adult                  2561 non-null   float64
 2   backdrop_path          1390 non-null   object 
 3   belongs_to_collection  206 non-null    object 
 4   budget                 2561 non-null   float64
 5   genres                 2561 non-null   object 
 6   homepage               173 non-null    object 
 7   id                     2561 non-null   float64
 8   original_language      2561 non-null   object 
 9   original_title         2561 non-null   object 
 10  overview               2510 non-null   object 
 11  popularity             2561 non-null   float64
 12  poster_path            2304 non-null   object 
 13  production_companies   2561 non-null   object 
 14  production_countries   2561 non-null   object 
 15  rele

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026  0.000  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092  0.000                               NaN                   NaN   
3  tt0116391  0.000                               NaN                   NaN   
4  tt0118694  0.000  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

        budget                                             genres homepage  \
0          NaN                                                NaN      NaN   
1 10000000.000  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2        0.000           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3        0.000  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4   150000.000  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

          id original_language   original_title  ...      revenue  runtime  \
0        NaN               NaN              NaN  ...          NaN      NaN   
1  62127.000                en  The Fantasticks  ...        0.000   86.000   
2 110977.000                en    For the Cause  ...        0.000  100.000   
3 442869.000                hi             Gang  ...        0.000  152.000   
4    843.000                cn             花樣年華  ... 12854953.000   99.000   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1  0.000         5.500     22.000           NaN  
2  0.000         5.100      8.000           NaN  
3  0.000         4.000      1.000           NaN  
4  0.000         8.115   2187.000            PG  

[5 rows x 26 columns]

**How many movies had at least some valid financial information (values > 0 for budget OR revenue)?**

In [59]:
# df2=df.loc[(df['Discount'] >= 1000) & (df['Discount'] <= 2000)]
df4 = df3.loc[df3['budget'] > 0]
df4.info()
df4.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 545 entries, 1 to 1306
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                545 non-null    object 
 1   adult                  545 non-null    float64
 2   backdrop_path          460 non-null    object 
 3   belongs_to_collection  101 non-null    object 
 4   budget                 545 non-null    float64
 5   genres                 545 non-null    object 
 6   homepage               68 non-null     object 
 7   id                     545 non-null    float64
 8   original_language      545 non-null    object 
 9   original_title         545 non-null    object 
 10  overview               543 non-null    object 
 11  popularity             545 non-null    float64
 12  poster_path            527 non-null    object 
 13  production_companies   545 non-null    object 
 14  production_countries   545 non-null    object 
 15  relea

imdb_id  adult                     backdrop_path  \
1   tt0113026  0.000  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg   
4   tt0118694  0.000  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
6   tt0119273  0.000  /f5C03doOWiauu37bToKXtpgP5bS.jpg   
11  tt0120467  0.000  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
12  tt0120626  0.000  /6xz9H7IMK3W21SNwAVmE9oW6YjX.jpg   

                                belongs_to_collection       budget  \
1                                                 NaN 10000000.000   
4                                                 NaN   150000.000   
6   {'id': 141086, 'name': 'Heavy Metal Collection... 15000000.000   
11                                                NaN   120000.000   
12                                                NaN  6105121.000   

                                               genres homepage        id  \
1   [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN 62127.000   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   843.000   
6   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...      NaN 16225.000   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...      NaN 19085.000   
12  [{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...      NaN 54380.000   

   original_language     original_title  ...      revenue  runtime  \
1                 en    The Fantasticks  ...        0.000   86.000   
4                 cn               花樣年華  ... 12854953.000   99.000   
6                 en   Heavy Metal 2000  ...        0.000   88.000   
11                en             Vulgar  ...    14904.000   87.000   
12                pt  Capitães de Abril  ...        0.000  123.000   

                                     spoken_languages    status  \
1   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
6   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
12  [{'english_name': 'Portuguese', 'iso_639_1': '...  Released   

                                              tagline                 title  \
1       Try to remember the first time magic happened       The Fantasticks   
4   Feel the heat, keep the feeling burning, let t...  In the Mood for Love   
6   not to survive the fight in the external world...      Heavy Metal 2000   
11   Everyone loves a clown... some more than others.                Vulgar   
12                                                NaN     Captains of April   

    video  vote_average vote_count certification  
1   0.000         5.500     22.000           NaN  
4   0.000         8.115   2187.000            PG  
6   0.000         6.092    180.000             R  
11  0.000         5.500     44.000             R  
12  0.000         6.400     52.000           NaN  

[5 rows x 26 columns]

In [60]:
df5 = df3.loc[df3['revenue'] > 0]
df5.info()
df5.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 446 entries, 4 to 1278
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                446 non-null    object 
 1   adult                  446 non-null    float64
 2   backdrop_path          427 non-null    object 
 3   belongs_to_collection  97 non-null     object 
 4   budget                 446 non-null    float64
 5   genres                 446 non-null    object 
 6   homepage               59 non-null     object 
 7   id                     446 non-null    float64
 8   original_language      446 non-null    object 
 9   original_title         446 non-null    object 
 10  overview               446 non-null    object 
 11  popularity             446 non-null    float64
 12  poster_path            445 non-null    object 
 13  production_companies   446 non-null    object 
 14  production_countries   446 non-null    object 
 15  relea

imdb_id  adult                     backdrop_path  \
4   tt0118694  0.000  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
10  tt0120263  0.000  /ynXVuylP8upazjz8lrqb1PEMkdR.jpg   
11  tt0120467  0.000  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
13  tt0120630  0.000  /sivJXJ8JvVVSRaaBnHgzOboNBRQ.jpg   
16  tt0120753  0.000  /3vUkQpnwbsFEQlMh9EORpvKJfpo.jpg   

                                belongs_to_collection       budget  \
4                                                 NaN   150000.000   
10                                                NaN        0.000   
11                                                NaN   120000.000   
13  {'id': 718551, 'name': 'Chicken Run Collection... 45000000.000   
16                                                NaN  8000000.000   

                                               genres homepage        id  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   843.000   
10  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...      NaN 34070.000   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...      NaN 19085.000   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...      NaN  7443.000   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...      NaN   318.000   

   original_language              original_title  ...       revenue  runtime  \
4                 cn                        花樣年華  ...  12854953.000   99.000   
10                sv  Sånger från andra våningen  ...     80334.000   98.000   
11                en                      Vulgar  ...     14904.000   87.000   
13                en                 Chicken Run  ... 224834564.000   84.000   
16                en    The Million Dollar Hotel  ...    105983.000  122.000   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
10  [{'english_name': 'Russian', 'iso_639_1': 'ru'...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
10                                                NaN   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   

                          title  video  vote_average vote_count certification  
4          In the Mood for Love  0.000         8.115   2187.000            PG  
10  Songs from the Second Floor  0.000         7.166    238.000           NaN  
11                       Vulgar  0.000         5.500     44.000             R  
13                  Chicken Run  0.000         6.742   4346.000             G  
16     The Million Dollar Hotel  0.000         5.921    280.000             R  

[5 rows x 26 columns]

In [61]:
## Excludes movies with 0's for budget and revenue
df6=df3.loc[(df3['revenue'] > 0) & (df3['budget'] >0)]
df6.info()
df6.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 354 entries, 4 to 1278
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                354 non-null    object 
 1   adult                  354 non-null    float64
 2   backdrop_path          347 non-null    object 
 3   belongs_to_collection  88 non-null     object 
 4   budget                 354 non-null    float64
 5   genres                 354 non-null    object 
 6   homepage               51 non-null     object 
 7   id                     354 non-null    float64
 8   original_language      354 non-null    object 
 9   original_title         354 non-null    object 
 10  overview               354 non-null    object 
 11  popularity             354 non-null    float64
 12  poster_path            354 non-null    object 
 13  production_companies   354 non-null    object 
 14  production_countries   354 non-null    object 
 15  relea

imdb_id  adult                     backdrop_path  \
4   tt0118694  0.000  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
11  tt0120467  0.000  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   
13  tt0120630  0.000  /sivJXJ8JvVVSRaaBnHgzOboNBRQ.jpg   
16  tt0120753  0.000  /3vUkQpnwbsFEQlMh9EORpvKJfpo.jpg   
17  tt0120755  0.000  /24DZfupDlhXeTchmcOkoGRhP5Vg.jpg   

                                belongs_to_collection        budget  \
4                                                 NaN    150000.000   
11                                                NaN    120000.000   
13  {'id': 718551, 'name': 'Chicken Run Collection...  45000000.000   
16                                                NaN   8000000.000   
17  {'id': 87359, 'name': 'Mission: Impossible Col... 125000000.000   

                                               genres homepage        id  \
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   843.000   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...      NaN 19085.000   
13  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...      NaN  7443.000   
16  [{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...      NaN   318.000   
17  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...      NaN   955.000   

   original_language            original_title  ...       revenue  runtime  \
4                 cn                      花樣年華  ...  12854953.000   99.000   
11                en                    Vulgar  ...     14904.000   87.000   
13                en               Chicken Run  ... 224834564.000   84.000   
16                en  The Million Dollar Hotel  ...    105983.000  122.000   
17                en    Mission: Impossible II  ... 546388105.000  123.000   

                                     spoken_languages    status  \
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
13  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
16  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
17  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
4   Feel the heat, keep the feeling burning, let t...   
11   Everyone loves a clown... some more than others.   
13  This ain't no chick flick. It's poultry in mot...   
16                    Everyone has something to hide.   
17                       Expect the impossible again.   

                       title  video  vote_average vote_count certification  
4       In the Mood for Love  0.000         8.115   2187.000            PG  
11                    Vulgar  0.000         5.500     44.000             R  
13               Chicken Run  0.000         6.742   4346.000             G  
16  The Million Dollar Hotel  0.000         5.921    280.000             R  
17    Mission: Impossible II  0.000         6.110   5754.000         PG-13  

[5 rows x 26 columns]

**Conclusion:**

* df4 (budget) has 545 entries
* df5 (revenue) has 446 entries
* df6 (both columns) has 354 entries 
* Total = 991 movies

In [67]:
# Math
545 + 446 

991

** 2. How many movies are there in each of the certification categories (G/PG/PG-13/R)?**

In [63]:
## 13 movies with financial info that have G rating
# Removes scientific notation
pd.set_option('display.float_format', lambda x: f'{x:.3f}')

df7 = df6.loc[df6['certification'] == 'G']
df7_mean = df7[["revenue","budget"]].mean()
print(df7_mean)
print('\n')
df7.info()


revenue   133216881.692
budget     44000000.000
dtype: float64


<class 'pandas.core.frame.DataFrame'>
Int64Index: 13 entries, 13 to 697
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                13 non-null     object 
 1   adult                  13 non-null     float64
 2   backdrop_path          13 non-null     object 
 3   belongs_to_collection  9 non-null      object 
 4   budget                 13 non-null     float64
 5   genres                 13 non-null     object 
 6   homepage               2 non-null      object 
 7   id                     13 non-null     float64
 8   original_language      13 non-null     object 
 9   original_title         13 non-null     object 
 10  overview               13 non-null     object 
 11  popularity             13 non-null     float64
 12  poster_path            13 non-null     object 
 13  production_companies   13 non-null     object

In [64]:
# 30 movies with PG rating
df8 = df6.loc[df6['certification'] == 'PG']
df8_mean = df8[["revenue","budget"]].mean()
print(df8_mean)
print('\n')
df8.info()

revenue   129080604.100
budget     51974907.800
dtype: float64


<class 'pandas.core.frame.DataFrame'>
Int64Index: 30 entries, 4 to 1150
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                30 non-null     object 
 1   adult                  30 non-null     float64
 2   backdrop_path          30 non-null     object 
 3   belongs_to_collection  11 non-null     object 
 4   budget                 30 non-null     float64
 5   genres                 30 non-null     object 
 6   homepage               10 non-null     object 
 7   id                     30 non-null     float64
 8   original_language      30 non-null     object 
 9   original_title         30 non-null     object 
 10  overview               30 non-null     object 
 11  popularity             30 non-null     float64
 12  poster_path            30 non-null     object 
 13  production_companies   30 non-null     object

In [65]:
# 117 movies with financial info and pg13 rating
df9 = df6.loc[df6['certification'] == 'PG-13']
df9_mean = df9[["revenue","budget"]].mean()
print(df9_mean)
print('\n')
df9.info()

revenue   111101802.316
budget     46627328.094
dtype: float64


<class 'pandas.core.frame.DataFrame'>
Int64Index: 117 entries, 17 to 1001
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                117 non-null    object 
 1   adult                  117 non-null    float64
 2   backdrop_path          116 non-null    object 
 3   belongs_to_collection  24 non-null     object 
 4   budget                 117 non-null    float64
 5   genres                 117 non-null    object 
 6   homepage               16 non-null     object 
 7   id                     117 non-null    float64
 8   original_language      117 non-null    object 
 9   original_title         117 non-null    object 
 10  overview               117 non-null    object 
 11  popularity             117 non-null    float64
 12  poster_path            117 non-null    object 
 13  production_companies   117 non-null    obje

In [66]:
# 142 movies with financial info and R rating
df10 = df6.loc[df6['certification'] == 'R']
df10_mean = df10[["revenue","budget"]].mean()
print(df10_mean)
print('\n')
df10.info()

revenue   52365899.563
budget    26863479.225
dtype: float64


<class 'pandas.core.frame.DataFrame'>
Int64Index: 142 entries, 11 to 910
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                142 non-null    object 
 1   adult                  142 non-null    float64
 2   backdrop_path          140 non-null    object 
 3   belongs_to_collection  28 non-null     object 
 4   budget                 142 non-null    float64
 5   genres                 142 non-null    object 
 6   homepage               20 non-null     object 
 7   id                     142 non-null    float64
 8   original_language      142 non-null    object 
 9   original_title         142 non-null    object 
 10  overview               142 non-null    object 
 11  popularity             142 non-null    float64
 12  poster_path            142 non-null    object 
 13  production_companies   142 non-null    object 